foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/mappings/OneToManyMapping.java

```diff
@@ -884,13 +918,42 @@ protected void objectAddedDuringUpdate(ObjectLevelModifyQuery query, Object obje
         if (requiresDataModificationEvents() || containerPolicy.requiresDataModificationEvents()){
             // In the uow data queries are cached until the end of the commit.
             if (query.shouldCascadeOnlyDependentParts()) {
```

Comment:
This implementation is so close to updateTargetForeignKeyPostUpdateSource_ObjectAdded() that I wish we could just have:

``` java
if (query.shouldCascadeOnlyDependentParts() && shouldDeferInsert()) {
    Object[] event = new Object[4];
    event[0] = ObjectAdded;
    event[1] = query;
    event[2] = objectAdded;
    event[3] = extraData;
    query.getSession().getCommitManager().addDataModificationEvent(this, event);
} else {
    updateTargetForeignKeyPostUpdateSource_ObjectAdded(query, objectAdded, extraData);
}
```
I don't understand the `OneToManyMapping.addTargetQuery` object enough tho

Reply:
I'd like to avoid code duplication as well but in this case it is non trivial and would very likely mean casts and ifs within the for loop - ie common parent for insert/datamodify queries is ModifyQuery which does not allow setting object on it, another thing is that `executeQuery(query, databaseRow)` apparently works differently for Insert and DataModify. Lambdas could safe us here but since we need this also in 2.6 (jdk7), that is not an option. At least not right now. We can improve this later...